src/DayColumn.js

```diff
@@ -275,9 +308,17 @@ class DayColumn extends React.Component {
     selector.on('selectStart', maybeSelect)
 
     selector.on('beforeSelect', box => {
+      const didNotClickOnAnEvent = !isEvent(findDOMNode(this), box)
+      if (
+        didNotClickOnAnEvent &&
+        this.props.components &&
+        this.props.components.beforeSelectIndicatorComponent
+      ) {
+        maybeSelect({ ...box, noMovementYet: true })
```

Comment:
there shouldn't be side effects in beforeSelect if possible

Reply:
Not sure I understand, can you explain a bit better what you mean by that?

Reply:
the idea of this callback is only to determine whether the selection should be handled, it should do any work, like storing coordinates. the `mousedown`, `click`, `selectStart`, etc handlers are for that

Reply:
Got it, so I can do something like:

```
selector.on('mousedown', box => {
      const didNotClickOnAnEvent = !isEvent(findDOMNode(this), box)
      if (
        didNotClickOnAnEvent &&
        this.props.components &&
        this.props.components.beforeSelectIndicatorComponent != null
      ) {
        maybeSelect({ ...box, noMovementYet: true })
      }
    })
```
?
I'll try that.

Reply:
Hmm, nope that doesn't work. I need that code to run on mouse down, but seems like `selector` never receives that event.